### PR TITLE
Docs for rate limiting

### PR DESCRIPTION
**Pull request by simonrozsival**
## What's this?
This PR adds docs for the rate limiting we will use in the syncing algorithm.

## Why do we want this?
We want everybody to understand the ideas behind the rate limiting implementation.

## How is it done?
I added a new chapter to the docs and updated the links.

## :squid: Permissions
🦑

### CODE REVIEW

File: docs/syncing/index.md [hunk 1]
```diff
@@ -12,6 +12,7 @@ The recommended reading order is this:
 7. [Transitions configuration](transitions-configuration.md)
 8. [Sample sync state](example.md)
 9. [Syncing tests](tests.md)
-10. [Background sync](bg-sync.md)
-11. [Enabling and disabling background sync](bg-sync-switching.md)
+10. [Rate limiting](rate-limiting.md)
+11. [Background sync](bg-sync.md)
 12. [Background Fetch on iOS](bg-fetch-ios.md)
+12. [Background Fetch on iOS](bg-fetch-android.md)
```
Comment (heytherewill) [inline docs/syncing/index.md]: Android*

File: docs/syncing/rate-limiting.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+Pull sync
+---------
+
+During pull sync we make 9 GET requests at once. We first try to allocate 9 slots from the minute bucket, and if that succeeds, we send the requests in "waves" of 3 requests. For each of the requests we wait until the seconds bucket gives us a free slot.
+
+Push sync
+---------
+
+During push sync we never make HTTP requests in parallel. We always check the minute bucket and if that succeeds, we wait until
```
Comment (sofokli1) [inline docs/syncing/rate-limiting.md]: It feels like this sentence is not finished

Comment (sofokli1) [reply docs/syncing/rate-limiting.md]: I get what it says, maybe the phrasing looks strange to me (or the lack of `.` in the end)

Comment (simonrozsival) [reply docs/syncing/rate-limiting.md]: Oh man, I definitely forgot to write the end of the sentence :D

Comment (simonrozsival) [reply docs/syncing/rate-limiting.md]: Oh man, I totally forgot to finish that sentence :D

**Comment by simonrozsival** [comment]
The section with Push sync is now finished.

**Pull request merged**